mirascope/openai/calls.py

```diff
@@ -185,3 +203,31 @@ async def stream_async(
                 self.call_params.response_format
             )
             yield openai_call_response_chunk
+
+    ############################## PRIVATE METHODS ###################################
+
+    def _setup_openai_kwargs(
+        self,
+        kwargs: dict[str, Any],
+    ) -> tuple[
+        dict[str, Any],
+        Optional[list[Type[OpenAITool]]],
+    ]:
+        """Overrides the `BaseCall._setup` for Anthropic specific setup."""
+        kwargs, tool_types = self._setup(kwargs, OpenAITool)
+        if self.call_params.response_format == {"type": "json_object"} and tool_types:
+            kwargs.pop("tools")
+        return kwargs, tool_types
+
+    def _maybe_update_messages(
```

Comment:
change function name to be `_update_message_if_json` so its more clear than maybe

Reply:
Done.